src/coreclr/jit/gentree.cpp

```diff
@@ -16604,6 +16604,18 @@ GenTree* Compiler::gtNewTempStore(
         valTyp      = lvaGetRealType(val->AsLclVar()->GetLclNum());
         val->gtType = valTyp;
     }
+
+    if (val->OperGet() == GT_BLK && val->AsBlk()->Addr()->OperGet() == GT_LCL_VAR)
+    {
+        GenTreeLclVarCommon* lclVar = val->AsBlk()->Addr()->AsLclVarCommon();
+        LclVarDsc* srcVarDsc = &lvaTable[lclVar->GetLclNum()];
+        if (srcVarDsc->lvRequiresSpecialCopy)
+        {
+            JITDUMP("Recording V%02u as an alias of the special-copy local V%02u\n", tmp, lclVar->GetLclNum());
+            GetSpecialCopyLocalsMap()->Emplace(tmp, lclVar->GetLclNum());
+        }
+    }
```

Comment:
I think this lookup should be done from call + arg back to local. Otherwise we are adding a requirement that all copies in the JIT need to be created in a specific way and that this table is maintained for them all. I don't think we need to do that.